packages/astro/src/core/cookies/cookies.ts

```diff
@@ -148,6 +150,13 @@ class AstroCookies implements AstroCookiesInterface {
 	 * @param options Options for the cookie, such as the path and security settings.
 	 */
 	set(key: string, value: string | Record<string, any>, options?: AstroCookieSetOptions): void {
+		if (this.#consumed) {
+			console.error(new Error(
+				"Astro.cookies.set() was called after the cookies have already been sent to the browser.\n" +
+				"This may have happened if this method was called in an imported component.\n" +
+				"Please make sure that Astro.cookies.set() is only called in the frontmatter of the main page."
+			));
```

Comment:
Should this be `console.warn` with a plain string instead according to the PR title? It feels like if we report an error, it usually means the request failed and returns 500 (or something)

Reply:
I addressed this in [`e29b456` (#9627)](https://github.com/withastro/astro/pull/9627/commits/e29b4560bd7d331a4c3e6a6e1ec3fea90d33eda6)

Reply:
That's much better, but we still need to provide some context as to where `cookies.set()` was called. Instantiating the error was one way to do it.

Reply:
<img width="742" alt="image" src="https://github.com/withastro/astro/assets/69170106/d69b0cca-83a8-48a2-a920-b83c35694bcd">